internal/k8s/service/resolver.go

```diff
@@ -321,9 +320,83 @@ func (r *backendResolver) findGlobalCatalogService(service *corev1.Service) (*Re
 	return nil, nil
 }
 
-// this will resolve a service based on our future CRD
-func (r *backendResolver) findConsulMeshService(name types.NamespacedName) {
-	mapped := r.mapper(name.Namespace)
+func (r *backendResolver) consulServiceForMeshService(ctx context.Context, namespacedName types.NamespacedName) (*ResolvedReference, error) {
+	var err error
+	var resolved *ResolvedReference
+
+	service, err := r.client.GetMeshService(ctx, namespacedName)
+	if err != nil {
+		r.logger.Trace("error retrieving mesh service", "error", err, "name", namespacedName.Name, "namespace", namespacedName.Namespace)
+		return nil, err
+	}
+	if service == nil {
+		return nil,
+			NewConsulResolutionError(fmt.Sprintf(
+				"kubernetes mesh service object (%q, %q) not found", namespacedName.Name, namespacedName.Namespace,
```

Comment:
We wound up just using the namespaced name directly in https://github.com/hashicorp/consul-api-gateway/pull/90 and below inside the retry. I'd like to see us settle on a convention for this

Reply:
yeah, that should be fine, the string output isn't much different, think the `NamespacedName`'s Stringer implementation outputs the equivalent of replacing: `(%q, %q)` with `{%s, %s}`